pkg/clients/s3/bucket.go

```diff
@@ -211,3 +212,28 @@ func UpdateBucketACL(ctx context.Context, client BucketClient, bucket *v1beta1.B
 	_, err := client.PutBucketAclRequest(config).Send(ctx)
 	return err
 }
+
+// CopyTag converts a local v1beta.Tag to an S3 Tag
+func CopyTag(tag *v1beta1.Tag) *s3.Tag {
+	return &s3.Tag{
+		Key:   aws.String(tag.Key),
+		Value: aws.String(tag.Value),
+	}
+}
+
+// CopyTags converts a list of local v1beta.Tags to S3 Tags
+func CopyTags(tags []v1beta1.Tag) []s3.Tag {
+	out := make([]s3.Tag, 0)
+	for _, one := range tags {
+		out = append(out, s3.Tag{Key: aws.String(one.Key), Value: aws.String(one.Value)})
+	}
+	return out
+}
+
+// SortS3TagSet stable sorts an external s3 tag list by the key and value.
+func SortS3TagSet(tags []s3.Tag) []s3.Tag {
```

Comment:
This operation is in-place and I see that in some places we use the returned value, like ina `switch` statement, and in some cases we ignore the returned value since we know that the array we supplied is changed.

I'd propose making a choice here; if the function makes changes in-place then no value should be returned or we can copy the array, make the operation and return the sorted copy of the array. Because as a user of this function, I wouldn't expect the operation to be in-place if a value is returned. It's fine either way but I think one should be chosen.

Reply:
Sounds good to me, I had written this way originally because I there were certain cases where I wanted the list returned, and others were it was sufficient if we sorted the list in place, but both cases can be simplified if we just always return a new list.